### PR TITLE
Added uuid to newsletter schema 

### PR DESCRIPTION
**Pull request by rshbhgrg**
refs https://github.com/TryGhost/Team/issues/1561

With multiple newsletters, unsubscribe links will also need to have a unique reference to the newsletter that the email is for, so that we can unsubscribe members from the particular newsletter automatically when they click on the link.
As our existing pattern for members is to use UUID as the external unique reference, this change adds UUID to newsletter schema and populates the existing newsletters with a UUID value.

- adds new `uuid` column to newsletter schema
- updates newsletter model to add default uuid

### CODE REVIEW

File: core/server/data/migrations/versions/4.46/2022-04-13-12-58-fill-uuid-for-newsletters.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Setting missing uuid values for existing newsletters');
+        const uuidV4 = uuid.v4();
+        const updatedRows = await knex('newsletters')
+            .where('uuid', null)
+            .update('uuid', uuidV4);
```
Comment (matthanley) [inline core/server/data/migrations/versions/4.46/2022-04-13-12-58-fill-uuid-for-newsletters.js:12]: Doesn't this try to give all rows the same UUID, so it'll fail if there's more than one? 🤔

Comment (rishabhgrg) [reply core/server/data/migrations/versions/4.46/2022-04-13-12-58-fill-uuid-for-newsletters.js:12]: Eugh yea good point, updating. I had in my mind only a single newsletter case here but we should definitely not rely on that.

Comment (rishabhgrg) [reply core/server/data/migrations/versions/4.46/2022-04-13-12-58-fill-uuid-for-newsletters.js]: Updated.

**Review by matthanley — APPROVED** [review]

File: core/server/data/migrations/versions/4.46/2022-04-13-12-57-add-uuid-column-to-newsletters.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const {createAddColumnMigration} = require('../../utils');
+
+module.exports = createAddColumnMigration('newsletters', 'uuid', {
+    type: 'string',
+    maxlength: 36,
+    nullable: true,
```
Comment (daniellockyer) [inline core/server/data/migrations/versions/4.46/2022-04-13-12-57-add-uuid-column-to-newsletters.js:6]: Should we be dropping the `nullable` on this field after we've populated it in the migration below?

**Review by SimonBackx — APPROVED** [review]
I also agree that it is best to drop the nullable after the backfill migration. But apart from that, everything looks okay 👌

File: core/server/data/migrations/versions/4.46/2022-04-13-12-57-add-uuid-column-to-newsletters.js (hunk 2 above)
Comment (rishabhgrg) [reply core/server/data/migrations/versions/4.46/2022-04-13-12-57-add-uuid-column-to-newsletters.js:6]: Does that mean we update the `schema` to have `nullable:false` and then just add a migration to drop it after ?

Comment (daniellockyer) [reply core/server/data/migrations/versions/4.46/2022-04-13-12-57-add-uuid-column-to-newsletters.js:6]: I believe so 🤔 https://github.com/TryGhost/Ghost/pull/14568/files#diff-18e7c5d15a806349f8a27507183762f3ef9d201415934325f2058f7d2c858c6b

**Review by daniellockyer — APPROVED** [review]

**Review by SimonBackx — APPROVED** [review]

**Pull request merged**